src/daemon/rpmostreed-os.c

```diff
@@ -710,6 +711,24 @@ os_handle_pkg_change (RPMOSTreeOS *interface,
 
   osname = rpmostree_os_get_name (interface);
 
+  /* Right now, we only use the fd list from the D-Bus message to transfer local
+   * packages, so there's no point in re-extracting them from fdlist based on
+   * the indices in local-packages. We just do a sanity check here that they are
+   * indeed the same length. If this assertion ever becomes false, then we'll
+   * have to extract. */
+  g_variant_dict_init (&options_dict, arg_options);
+  if (g_variant_dict_contains (&options_dict, "local-packages"))
+    {
+      g_autoptr(GVariant) fds =
+        g_variant_dict_lookup_value (&options_dict, "local-packages",
+                                     G_VARIANT_TYPE ("ah"));
+      g_assert (fds);
+      g_assert (g_unix_fd_list_get_length (fdlist) ==
```

Comment:
While this method is only available to privileged callers, I think we should return an error here rather than assert - as a general rule, callers of DBus methods shouldn't be able to trivially cause the server to abort.